backend/src/Types/App.hs

```diff
@@ -1,17 +1,21 @@
 module Types.App where
 
-import Data.Pool (Pool)
+import Database.Esqueleto.Experimental (SqlPersistT)
+import Database.Esqueleto.Experimental qualified as SQL
 import Database.Redis qualified as Redis
-import Database.SQLite.Simple qualified as SQL
 import Servant (Handler)
 import System.Log.FastLogger (LogStr, TimedFastLogger, ToLogStr (..))
 
-data Env = Env {dbPool :: Pool SQL.Connection, redisPool :: Redis.Connection, logger :: TimedFastLogger}
+data Env = Env {dbPool :: SQL.ConnectionPool, redisPool :: Redis.Connection, logger :: TimedFastLogger}
 
 type AppM = ReaderT Env Handler
 
 nt :: Env -> AppM a -> Handler a
-nt s x = runReaderT x s
+nt env server = runReaderT server env
 
+runDb :: SqlPersistT IO a -> AppM a
```

Comment:
Helper function to run SQL actions inside the AppM monad. All queries inside the action are executed in a single transaction.

Reply:
✨